source/cppfs/include/cppfs/cppfs.h

```diff
@@ -29,5 +29,26 @@ enum FilePermissions
     Sticky     = 01000
 };
 
+/**
+*  @brief
+*    Type of event on the file system
+*/
+enum FileEvent
+{
+    FileCreated  = 0x01, ///< A file or directory has been created
+    FileRemoved  = 0x02, ///< A file or directory has been removed
+    FileModified = 0x04  ///< A file or directory has been modified
+};
+
+/**
+*  @brief
+*    Recursive mode for operation that can run recursively or non-recursively
+*/
+enum RecursiveMode
```

Comment:
Is there any chance that there will be more than one recursive mode?
As of now, this enum seems to be a boolean.

Reply:
Yes, this is a boolean, but with a semantic. I don't want to use a boolean for this kind of flags any more, because when I read code like fs.createFile(filename, true, false, true) ... what does it mean? If I read fs.createFile(filename, Recursive, NoTruncate, Create), I immediately get the meaning. So this is to increase comprehension when reading the code. Do you have a better way of doing this?

Reply:
I now better ways, but they are not feasible in C++: named arguments. Of cause, we can implement them but they come with a sensible performance overhead or hard-to-use syntax.

Either way, I would argue for better readable code and vote for using enums over booleans for this case.